base/lttemplates.dtx

```diff
@@ -351,12 +351,89 @@
+%    \meta{template} they are simply ignored and stored in the
+%    tokenlist \cs{UnusedTemplateKeys}
+%    without generating an error. This way it is possible to
+%    apply the same key/val list specified by the user on a
+%    document-level command or environment to several templates, which
+%    is useful, if their combination makes up that command or
+%    environment. As a variation of that, you can use this key/value
+%    list the first time, and for the next template use what remains in
+%    \cs{UnusedTemplateKeys}. The final processing step could then be
+%    \cs{SetTemplateKeys}, which unconditionally attempts to set the
+%    \meta{keyvals} received in its third argument. That command
+%    complains if any of them are unknown keys. Alternatively, you
+%    could check if \cs{UnusedTemplateKeys} is empty.\footnote{Using
+%    \cs{SetTemplateKeys} exposes the inner structure of the template
+%    keys when generating an error. This is something one may want to
+%    avoid as it can be confusing to the user, especially if several
+%    templates are involved. In that
+%    case use  \cs{SetKnownTemplateKeys} and afterwards check if
+%    \cs{UnusedTemplateKeys} is empty, and if not generate your own
+%    error message.}
+
+%    For example, a list, such as \env{enumerate}, is made up from a
+%    \texttt{blockenv}, \texttt{block}, \texttt{list}, and a
+%    \texttt{para} template and in the single user-supplied optional
+%    argument of \env{enumerate} key/values for either of
+%    the templates might get specified. In fact, in that particular example,
+%    that key/value list is also saved and then applied to each
+%    \cs{item} which is implemented through an \texttt{item}
+%    template. This way, one can specify one-off settings to all items
+%    of a single whole list (on the environment level), as well as to
+%    individual items within a list (by specifying them in the
+%    optional argument of an \cs{item}).  With
+%    \cs{SetKnownTemplateKeys} and \cs{SetTemplateKeys} working
+%    together, it is possible to provide this flexibility and still
+%    alert the user if one of their keys is misspelled.
+%
+%    On the other hand you may want to allow for
+%    \enquote{misspellings} without generating an error or a
+%    warning. For example, if you define an template that only accepts
+%    few keys, you might just want to ignore anything specified in the
+%    source if you use this template in place of a different one,
+%    without the need to alter the document source. Or you might just
+%    generate a warning message, which is easy, given that the unused
+%    key/values are available in the \cs{UnusedTemplateKeys} variable.
+%
+%    The \cs{SetKnownTemplateKeys} and \cs{SetTemplateKeys} commands
+%    are only supported within the code of a template; using them
+%    elsewhere has unpredictable results.  If they are used together
+%    with \cs{AssignTemplateKeys} then this command should come first
```

Comment:
Unclear: 
Which should come first?

Reply:
the latter command?   I want to avoid writing \AssignTemplateKeys twice

Reply:
Yes, better to be more precise than just "this".